packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -224,9 +237,45 @@ class DebugBundleFlutterAssets extends Target {
             resource.release();
           }
         }));
-    } catch (err, st){
+    } catch (err, st) {
       throw Exception('Failed to copy assets: $st');
     }
+    // Copy Info.plist template.
+    assetDirectory.parent.childFile('Info.plist')
+      ..createSync()
+      ..writeAsStringSync(r'''
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleDevelopmentRegion</key>
+	<string>$(DEVELOPMENT_LANGUAGE)</string>
+	<key>CFBundleExecutable</key>
+	<string>$(EXECUTABLE_NAME)</string>
+	<key>CFBundleIconFile</key>
+	<string></string>
+	<key>CFBundleIdentifier</key>
+	<string>io.flutter.flutter.app</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>6.0</string>
+	<key>CFBundleName</key>
+	<string>$(PRODUCT_NAME)</string>
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
+	<key>CFBundleVersion</key>
+	<string>1</string>
+	<key>LSMinimumSystemVersion</key>
+	<string>$(MACOSX_DEPLOYMENT_TARGET)</string>
+	<key>NSMainStoryboardFile</key>
+	<string>Main</string>
+	<key>NSPrincipalClass</key>
```

Comment:
This looks like an application Info.plist, not a Framework one.  CFBundleIconFile, NSPrincipalClass, NSMainStoryboardFile keys shouldn't be there, and the CFBundlePackageType should be FMWK.
The one I just generated:
```
$ plutil -p Info.plist 
{
  "CFBundleDevelopmentRegion" => "$(DEVELOPMENT_LANGUAGE)"
  "CFBundleExecutable" => "$(EXECUTABLE_NAME)"
  "CFBundleIdentifier" => "$(PRODUCT_BUNDLE_IDENTIFIER)"
  "CFBundleInfoDictionaryVersion" => "6.0"
  "CFBundleName" => "$(PRODUCT_NAME)"
  "CFBundlePackageType" => "$(PRODUCT_BUNDLE_PACKAGE_TYPE)"
  "CFBundleShortVersionString" => "1.0"
  "CFBundleVersion" => "$(CURRENT_PROJECT_VERSION)"
  "NSHumanReadableCopyright" => "Copyright © 2019 <etc> . All rights reserved."
}
```
And it generated from Xcode as:
```
{
  "BuildMachineOSBuild" => "18G87"
  "CFBundleDevelopmentRegion" => "en"
  "CFBundleExecutable" => "TestFramework"
  "CFBundleIdentifier" => "com.google.TestFramework"
  "CFBundleInfoDictionaryVersion" => "6.0"
  "CFBundleName" => "TestFramework"
  "CFBundlePackageType" => "FMWK"
  "CFBundleShortVersionString" => "1.0"
  "CFBundleSupportedPlatforms" => [
    0 => "MacOSX"
  ]
  "CFBundleVersion" => "1"
  "DTCompiler" => "com.apple.compilers.llvm.clang.1_0"
  "DTPlatformBuild" => "11M382q"
  "DTPlatformVersion" => "GM"
  "DTSDKBuild" => "19A526e"
  "DTSDKName" => "macosx10.15"
  "DTXcode" => "1100"
  "DTXcodeBuild" => "11M382q"
  "NSHumanReadableCopyright" => "Copyright © 2019 <etc> . All rights reserved."
}
```

Reply:
Ahh, thanks! I'll update the template

Reply:
Updated to use the Info.plist from a cocoa framework project. I don't think we need/want the copyright key, correct?